prow/tide/tide.go

```diff
@@ -103,17 +112,78 @@ func NewController(ghc *github.Client, kc *kube.Client, ca *config.Agent, gc *gi
 	}
 }
 
+// org/repo -> number -> pr
+func byRepoAndNumber(prs []PullRequest) map[string]map[int]PullRequest {
+	m := make(map[string]map[int]PullRequest)
+	for _, pr := range prs {
+		r := string(pr.Repository.NameWithOwner)
+		if _, ok := m[r]; !ok {
+			m[r] = make(map[int]PullRequest)
+		}
+		m[r][int(pr.Number)] = pr
+	}
+	return m
+}
+
+// Returns expected status state and description.
+// TODO(spxtr): Useful information such as "missing label: foo."
+func expectedStatus(pr PullRequest, pool map[string]map[int]PullRequest) (githubql.StatusState, string) {
+	if _, ok := pool[string(pr.Repository.NameWithOwner)][int(pr.Number)]; !ok {
+		return githubql.StatusStatePending, tideStatusNotInPool
+	}
+	return githubql.StatusStateSuccess, tideStatusInPool
+}
+
+func (c *Controller) setStatuses(all, pool []PullRequest) error {
+	poolM := byRepoAndNumber(pool)
+	for _, pr := range all {
+		wantState, wantDesc := expectedStatus(pr, poolM)
+		var actualState githubql.StatusState
+		var actualDesc string
+		for _, ctx := range pr.Commits.Nodes[0].Commit.Status.Contexts {
+			if string(ctx.Context) == tideStatusContext {
+				actualState = ctx.State
+				actualDesc = string(ctx.Description)
+			}
+		}
+		if wantState != actualState || wantDesc != actualDesc {
+			if err := c.ghc.CreateStatus(
+				string(pr.Repository.Owner.Login),
+				string(pr.Repository.Name),
+				string(pr.HeadRef.Target.OID),
+				github.Status{
+					Context:     tideStatusContext,
+					State:       string(wantState),
+					Description: wantDesc,
+					TargetURL:   c.ca.Config().Tide.TargetURL,
```

Comment:
We could make the details link specify the individual tide pool and PR number by appending parameters to the URL. We could then make the tide front end highlight the correct tide pool and PR and possibly hide other pools. I think that would be helpful for new users of Tide or even for more regular users when there are many tide pools displayed.
Probably outside the scope of this PR though.

Reply:
Agreed. Another useful thing to do is analyze the PR and change the description to say things like "missing label: lgtm" and so on. I'd like to do those in a future PR.